src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -455,33 +455,71 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
                                     keyField = table.getKeyFieldName();
                                     keyFieldIndex = table.getKeyFieldIndex(fieldsFoundInZip);
                                     keyValue = csvReader.get(keyFieldIndex);
-                                    stopCodeMissingFromFirstTable = true;
-                                }
-                            } else {
-                                // Check whether stop_code exists for the subsequent files.
-                                String firstStopCodeValue = csvReader.get(getFieldIndex(fieldsFoundInZip, "stop_code"));
-                                if (stopCodeMissingFromFirstTable && !"".equals(firstStopCodeValue)) {
-                                    // If stop_code was missing from the first file and exists for
-                                    // the second, we consider that a failing error.
+                                    if (feedIndex == 0) {
+                                        stopCodeMissingFromFirstFeed = true;
+                                    }
+                                    if (feedIndex == 1 && !stopCodeMissingFromFirstFeed) {
+                                        mergeFeedsResult.failed = true;
+                                        mergeFeedsResult.errorCount++;
+                                        mergeFeedsResult.failureReasons.add(
+                                            String.format(
+                                                "If stop_code is provided for some stops (for those with location_type = " +
+                                                    "empty or 0), all stops must have stop_code values. The merge process " +
+                                                    "found %d stops that were incorrectly missing stop_code values.",
+                                                stopsMissingStopCodeCount
+                                            )
+                                        );
+                                    }
+                                } else if (stopsMissingStopCodeCount > 0) {
+                                    // If some, but not all, stops are missing stop_code, the merge feeds job must fail.
                                     mergeFeedsResult.failed = true;
                                     mergeFeedsResult.errorCount++;
                                     mergeFeedsResult.failureReasons.add(
-                                        "If one stops.txt file contains stop_codes, both feed versions must stop_codes.");
+                                        String.format(
+                                            "If stop_code is provided for some stops (for those with location_type = " +
+                                                "empty or 0), all stops must have stop_code values. The merge process " +
+                                                "found %d stops that were incorrectly missing stop_code values.",
```

Comment:
The comment might be improved by expressing the number of bad stops out of total stops. For extra extra credit, also omit stations somehow.